src/c++/perf_analyzer/infer_context.cc

```diff
@@ -251,12 +242,20 @@ InferContext::AsyncCallbackFuncImpl(cb::InferResult* result)
       thread_stat_->cb_status_ = result_ptr->Id(&request_id);
       const auto& it = async_req_map_.find(request_id);
       if (it != async_req_map_.end()) {
-        thread_stat_->request_timestamps_.emplace_back(std::make_tuple(
-            it->second.start_time_, end_time_async, it->second.sequence_end_,
-            it->second.delayed_));
-        infer_backend_->ClientInferStat(&(thread_stat_->contexts_stat_[id_]));
-        thread_stat_->cb_status_ = ValidateOutputs(result);
-        async_req_map_.erase(request_id);
+        bool is_final_response{false};
+        thread_stat_->cb_status_ =
```

Comment:
This might overwrite a bad status from line 242.  I think every setting of cb_status_ needs to be followed immediately by a return if not ok.

Reply:
Good catch--added those checks above 👍

Reply:
Backtracking here--ran into an issue where returning right after a ClientBackend API returned an error caused unexpected behavior (CI failure). I'm only going to return when failure happens for the API I added (`IsFinalResponse()`) for now. Need deeper understanding of the async callback to make the changes you suggested.